sae_lens/training/activations_store.py

```diff
@@ -655,6 +660,198 @@ def save(self, file_path: str):
+        total_training_tokens: int,
+        store_batch_size_prompts: int,
+        train_batch_size_tokens: int,
+        prepend_bos: bool,
+        normalize_activations: str,
+        device: torch.device,
+        dtype: str,
+        cached_activations_path: str | list[str] | None = None,
+        model_kwargs: dict[str, Any] | None = None,
+        autocast_lm: bool = False,
+        dataset_trust_remote_code: bool | None = None,
+    ):
+        self.grid_wise = grid_wise
+        self.current_epoch = -1
+
+        assert cached_activations_path is not None
+        if isinstance(cached_activations_path, str):
+            cached_activations_path = [cached_activations_path]
+
+        super().__init__(
+            model=model,
+            dataset=dataset,
+            streaming=streaming,
+            hook_name=hook_name,
+            hook_layer=hook_layer,
+            hook_head_index=hook_head_index,
+            context_size=1,
+            d_in=d_in,
+            n_batches_in_buffer=1,
+            total_training_tokens=total_training_tokens,
+            store_batch_size_prompts=1,
+            train_batch_size_tokens=train_batch_size_tokens,
+            prepend_bos=prepend_bos,
+            normalize_activations=normalize_activations,
+            device=device,
+            dtype=dtype,
+            cached_activations_path=cached_activations_path,
+            model_kwargs=model_kwargs,
+            autocast_lm=autocast_lm,
+            dataset_trust_remote_code=dataset_trust_remote_code,
+        )
+
+    @classmethod
+    def from_config(
+        cls,
+        model: HookedRootModule,
+        cfg: DRCSAERunnerConfig,
+        override_dataset: HfDataset | None = None,
+    ) -> "DRCActivationsStore":
+        return cls(
+            grid_wise=cfg.grid_wise,
+            model=model,
+            dataset=override_dataset or cfg.dataset_path,
+            streaming=cfg.streaming,
+            hook_name=cfg.hook_name,
+            hook_layer=cfg.hook_layer,
+            hook_head_index=cfg.hook_head_index,
+            context_size=cfg.context_size,
+            d_in=cfg.d_in,
+            n_batches_in_buffer=cfg.n_batches_in_buffer,
+            total_training_tokens=cfg.training_tokens,
+            store_batch_size_prompts=cfg.store_batch_size_prompts,
+            train_batch_size_tokens=cfg.train_batch_size_tokens,
+            prepend_bos=cfg.prepend_bos,
+            normalize_activations=cfg.normalize_activations,
+            device=torch.device(cfg.act_store_device),
+            dtype=cfg.dtype,
+            cached_activations_path=cfg.cached_activations_path,
+            model_kwargs=cfg.model_kwargs,
+            autocast_lm=cfg.autocast_lm,
+            dataset_trust_remote_code=cfg.dataset_trust_remote_code,
+        )
+
+    @torch.no_grad()
+    def get_buffer(
+        self, n_batches_in_buffer: int, raise_on_epoch_end: bool = False
+    ) -> torch.Tensor:
+        if "acts_ds" not in self.__dict__:
```

Comment:
```suggestion
        if not hasattr(self, "acts_ds"):
```

though functionally the same, doens't rely on implementation detail of arbitrary attrs being in `__dict__`.